include/swift/SILOptimizer/Differentiation/AdjointValue.h

```diff
@@ -38,10 +40,25 @@ enum AdjointValueKind {
 
   /// A concrete SIL value.
   Concrete,
+
+  /// A special adjoint, made up of 2 adjoints -- a base adjoint and an element
```

Comment:
Do we really need to provide a "clarification" here? This might confuse the reader. What is confusing is `base adjoint` term. It's not quite so. We're essentially using only type (and index there). Also, do we really need to use `Add` here? I think the semantics is essentially `(0, ..., 0, el', 0, ..., 0)`, so the non-trivial combination of zero and aggregate adjoints.

Reply:
Side-stepping the naming suggestions for a second, what I understood about the semantics of this new adjoint kind is -
1. It is a combination (by addition) of 2 adjoints, let's call them LHS and RHS for now. 
2. The LHS will always be a zeroed aggregate adjoint.
3. The RHS will be the adjoint for one of the fields of LHS. 

Symbolically, we have -
```
AddElement(LHS, #field_or_index, RHS) == (materialization) ==> AddElement.#field_or_index += RHS
```
There might be questions/comments about how we're accumulating this new adjoint with others, but I think establishing a common understanding of its semantics first might be helpful 🙂.